src/turn.erl

```diff
@@ -467,64 +483,100 @@ code_change(_OldVsn, StateName, State, _Extra) ->
+                               maps:put(Addr, TRef, Acc)
+                            end,
+                            State#state.permissions,
+                            Addrs),
+            {ok, State#state{permissions = Perms}};
+        {false, _} ->
+            {error, 443};
+        {_, true} ->
+            {error, 403}
     end.
 
 send(State, Pkt) when is_binary(Pkt) ->
     SockMod = State#state.sock_mod,
     Sock = State#state.sock,
     if SockMod == gen_udp ->
-	    {Addr, Port} = State#state.addr,
-	    gen_udp:send(Sock, Addr, Port, Pkt);
+           {Addr, Port} = State#state.addr,
+           gen_udp:send(Sock, Addr, Port, Pkt);
        true ->
-	    case SockMod:send(Sock, Pkt) of
-		ok ->
-		    ok;
-		_  ->
-		    ?LOG_INFO("Cannot respond to client: Connection closed"),
-		    exit(normal)
-	    end
+           case SockMod:send(Sock, Pkt) of
+               ok ->
+                   ok;
+               _ ->
+                   ?LOG_INFO("Cannot respond to client: Connection closed"),
+                   exit(normal)
+           end
     end;
 send(State, Msg) ->
     ?LOG_DEBUG(#{verbatim => {"Sending:~n~s", [stun_codec:pp(Msg)]}}),
     Key = State#state.key,
     case Msg of
-	#stun{class = indication} ->
-	    send(State, stun_codec:encode(Msg)),
-	    State;
-	#stun{class = response} ->
-	    Pkt = stun_codec:encode(Msg, Key),
-	    send(State, Pkt),
-	    State#state{last_trid = Msg#stun.trid,
-			last_pkt = Pkt};
-	_ ->
-	    send(State, stun_codec:encode(Msg, Key)),
-	    State
+        #stun{class = indication} ->
+            send(State, stun_codec:encode(Msg)),
+            State;
+        #stun{class = response} ->
+            Pkt = stun_codec:encode(Msg, Key),
+            send(State, Pkt),
+            State#state{last_trid = Msg#stun.trid, last_pkt = Pkt};
+        _ ->
+            send(State, stun_codec:encode(Msg, Key)),
+            State
+    end.
+
+send_payload_to_client(Payload, Peer, State) ->
+    {PeerAddr, _PeerPort} = Peer,
+    case {maps:find(PeerAddr, State#state.permissions), maps:find(Peer, State#state.peers)} of
+        {{ok, _}, {ok, Channel}} ->
+            TurnMsg = #turn{channel = Channel, data = Payload},
+            State1 = count_rcvd(State, Payload),
+            send(State1, TurnMsg);
+        {{ok, _}, error} ->
+            Seq = State#state.seq,
+            Ind = #stun{class = indication,
+                        method = ?STUN_METHOD_DATA,
+                        trid = Seq,
+                        'XOR-PEER-ADDRESS' = [Peer],
+                        'DATA' = Payload},
+            State1 = count_rcvd(State, Payload),
+            send(State1#state{seq = Seq + 1}, Ind);
+        {error, _} ->
+            State
+    end.
+
+is_stun_packet(<<Head:8, _Tail/binary>>) when Head < 2 ->
+    true;
+is_stun_packet(_Pkt) ->
+    false.
+
+maybe_send_to_ice_bin(#state{peer_pid = PeerPid}, Payload) ->
```

Comment:
This lib shouldn't have dependency to ICE plugin. Could we rename this function somehow?